File: CS110/Projects/project1/great_circle.py
import math
import stdio
import sys

# Inputs
x1 = math.radians(float(sys.argv[1])) # (x1, y1)
y1 = math.radians(float(sys.argv[2]))

x2 = math.radians(float(sys.argv[3])) # (x2, y2)
y2 = math.radians(float(sys.argv[4]))

# Calculations
distance = 6359.83 * math.acos(math.sin(x1) * math.sin(x2) + math.cos(x1) * math.cos(x2) * math.cos(y1 - y2))

# Output
stdio.writeln(str(distance))

File: CS210/Exercises/exercise1/src/GreatCircle.java
import stdlib.StdOut;

public class GreatCircle {
    // Entry point.
    public static void main(String[] args) {
        // Accept x1 (double), y1 (double), x2 (double), and y2 (double) as command-line arguments.
		double x1 = Math.toRadians(Double.parseDouble(args[0]));
		double y1 = Math.toRadians(Double.parseDouble(args[1]));
		double x2 = Math.toRadians(Double.parseDouble(args[2]));	
		double y2 = Math.toRadians(Double.parseDouble(args[3]));

        // Convert the angles to radians.
		

        // Calculate great-circle distance d.
		double distance = 6359.83 * Math.acos(Math.sin(x1) * Math.sin(x2) + Math.cos(x1) * Math.cos(x2) * Math.cos(y1- y2));

        // Write d to standard output.
        System.out.println(distance);
    }
}
